### PR TITLE
feat: Athens 3 support

### PR DESCRIPTION
**Pull request by fadeev**
* added a workflow to publish to npm
* updated the readme to include examples on how to import different parts of the toolkit
* refactored helpers
* added a `trackCCTX` function that can be used whenever CCTX are broadcasted to get up-to-date information on the CCTX status
* updated the `balances` task to show Bitcoin and ZRC20 balances
* added a `btc` task to send tBTC (between BTC addresses and to ZetaChain by sending to a TSS address)
* added a `send` task to send ZETA between connected chain using the Connector contract
* removed the `gasFee = gasFee * 2;` workaround

### CODE REVIEW

File: contracts/EthZetaMock.sol [hunk 1]
```diff
@@ -6,7 +6,7 @@ import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/utils/Context.sol";
 
 contract ZetaEthMock is ERC20("Zeta", "ZETA") {
-    constructor(address creator, uint256 initialSupply) {
-        _mint(creator, initialSupply * (10 ** uint256(decimals())));
+    constructor(uint256 initialSupply) {
```
Comment (andresaiello) [inline contracts/EthZetaMock.sol:9]: why this change? we change it to support creator in the protocol version because is easy to generate fancy addresses (ownership belongs to the factory)

Comment (fadeev) [reply contracts/EthZetaMock.sol]: reverted https://github.com/zeta-chain/toolkit/pull/10/commits/e35a1ea3f054223087b4b63d51a9881703a3d71b

File: tasks/sendBTC.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  const endpoint = `${ENDPOINT}/decode`;
+
+  const p1 = await fetch(endpoint, {
+    body: JSON.stringify({ tx }),
+    method: "POST",
+  });
+  return await p1.json();
+};
+
+async function fetchUtxos(address: string): Promise<UTXO[]> {
```
[the remark this answers is not in the document]
Comment (fadeev) [reply tasks/sendBTC.ts]: https://github.com/zeta-chain/toolkit/pull/10/commits/c93fee5fc67ac43638686753f1cc27327e13d123

We should probably add it to the linter.

[the remark this answers is not in the document]
Comment (fadeev) [reply tasks/sendBTC.ts]: This should enforce arrow functions: https://github.com/zeta-chain/toolkit/pull/10/commits/620943bd59d425caa785c8506ff1ab86a5184f04

File: helpers/tx.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import axios from "axios";
+import moment from "moment";
+import ora from "ora";
+import WebSocket from "ws";
+
+const URL = {
+  API: "https://zetachain-athens.blockpi.network/lcd/v1/public/zeta-chain/crosschain",
```
Comment (btroo) [inline helpers/tx.ts:7]: should this pull from a package like @zetachain/networks?

File: tasks/balances.ts [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
 import { task } from "hardhat/config";
 import { HardhatRuntimeEnvironment } from "hardhat/types";
+import ora from "ora";
+import * as ecc from "tiny-secp256k1";
+
+declare const hre: any;
+
+dotenv.config();
+
+const BTC_API = "https://blockstream.info/testnet/api";
```
Comment (btroo) [inline tasks/balances.ts:17]: should this bitcoin network/api information be in a config / package?

File: tasks/balances.ts [hunk 5]
```diff
@@ -27,69 +39,122 @@ const balancesError = `
...
+        return `${b} ${denom}`;
+      }
     } catch (error) {}
-  };
+  });
+
+  const result = await Promise.all(promises);
+
+  // tBTC ZRC-20 balance
+  const btcZRC20 = "0x65a45c57636f9BcCeD4fe193A602008578BcA90b"; // use getAddress("zrc20", "btc_testnet") when available
```
Comment (btroo) [inline tasks/balances.ts:119]: maybe any comments that need actions could be marked with `TODO:`?

File: tasks/sendBTC.ts [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+import ECPairFactory from "ecpair";
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import * as ecc from "tiny-secp256k1";
+
+dotenv.config();
+
+const TESTNET = bitcoin.networks.testnet;
+
+const API = "https://blockstream.info/testnet/api";
```
Comment (btroo) [inline tasks/sendBTC.ts:13]: same comment here on if this api/testnet should be extracted into a config/package

File: tasks/sendBTC.ts [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+  return response.json();
+};
+
+const makeTransaction = async (
+  to: string,
+  key: any,
+  amount: any,
+  utxos: any,
+  address: string,
+  memo: string = ""
```
Comment (btroo) [inline tasks/sendBTC.ts:43]: later, it would be cool if we have a tool to help you construct the memo for btc (cli tool like this, or a helper function)!

**Review by btroo — APPROVED** [review]
LGTM! a couple questions/comments but nothing major. also, reviewed assuming the generated files are 👌

File: tasks/balances.ts (hunk 4 above)
Comment (fadeev) [reply tasks/balances.ts:17]: Done: https://github.com/zeta-chain/toolkit/pull/10/commits/472f54f1f02499d7e729af4bc0e8301a36b3b02c

I can't publish an updated version of `@zetachain/publish` for some reason, though:

https://github.com/zeta-chain/networks/actions/runs/5518574540/jobs/10062714440

So for now this works only when linking the package directly.

File: tasks/balances.ts (hunk 5 above)
Comment (fadeev) [reply tasks/balances.ts]: https://github.com/zeta-chain/toolkit/pull/10/commits/e9324e92357b1657d8cb8b47bda47ac6215bb45f

File: tasks/sendBTC.ts (hunk 6 above)
Comment (fadeev) [reply tasks/sendBTC.ts]: Added tBTC to `@zetachain/networks`: https://github.com/zeta-chain/toolkit/pull/10/commits/11be332dd7ef2c06ef73a4b011bff118c62d4834

As for `TESTNET`, these values are published as part of the `bitcoinjs-lib` package, so I don't think we need to publish it ourselves.

File: tasks/sendBTC.ts (hunk 7 above)
Comment (fadeev) [reply tasks/sendBTC.ts:41]: https://github.com/zeta-chain/toolkit/issues/16

**Comment by fadeev** [comment]
Thanks for reviewing, @btroo!

File: helpers/tx.ts (hunk 3 above)
Comment (fadeev) [reply helpers/tx.ts]: Done: https://github.com/zeta-chain/toolkit/pull/10/commits/472f54f1f02499d7e729af4bc0e8301a36b3b02c

I can't publish an updated version of `@zetachain/publish` for some reason, though:

https://github.com/zeta-chain/networks/actions/runs/5518574540/jobs/10062714440

So for now this works only when linking the package directly.

File: tasks/balances.ts (hunk 4 above)
Comment (fadeev) [reply tasks/balances.ts]: https://github.com/zeta-chain/toolkit/pull/10/commits/3d1da4c5bdbd5e915f50ceafa86e4a40d2961dad

**Review by btroo — APPROVED** [review]

**Review by btroo — APPROVED** [review]
